File: leetcode/删除排序数组中的重复项.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    for (let i = 0; i < nums.length;) {
      if (nums[i] === nums [i + 1]) nums.splice(i, 1);
      else i ++;
    }
  return nums.length;
};

File: leetcode/删除排序数组中的重复项.go
func removeDuplicates(nums []int) int {
	for i:=1; i<len(nums); i++ {
        temp := nums[i-1] // 前一项与当前项进行对比
		if temp == nums[i] {
			nums = append(nums[:i],nums[i+1:]...)
            i-- // 继续用当前项对比，i保持不变，抵消掉循环中的i++，故此i--
        }
	}
	return len(nums)
}
